### PR TITLE
SVG Refactor + Icon Hotfix

### LINKED ISSUE #77: 🐛 Bug: Difficulty icon bleeds outside of customize card on mobile screens
### Expected Behavior 🧭

The difficulty icon should not bleed outside of the customize card on mobile screens. 

### Current Behavior 🔍 

The ifficulty icon bleeds outside of customize card on mobile screens

image

### Steps to Reproduce 🔢 

View Material math on mobile screens

**Acceptence Criteria for Fix ✅**

(Please write your answer here.)

- [ ] The fixed version should like like the card on desktop: 
image

### Possible Solution 🛠️ 

The icon sizes should probably be reduced

### PR DESCRIPTION
**Pull request by diracs-delta**
# Summary

- call `vue-svg-loader` via webpack to import SVGs as Vue components. this also optimizes SVGs via `svgo` in the build process, greatly improving load times of SVG assets. (will extend this to other SVG assets later)
- extensively refactor PracticeCard to use the QCard component shipped with Quasar.
- fix spacing and margin issues, resolves #77 
- preserves current UI and appearance

# Preview

### CODE REVIEW

File: src/components/PracticeCard.vue [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  </div>
+</q-card>
+</template>
+
+<script lang="ts">
+import { PracticeGetters, PracticeActions } from "../store/practice/practice";
+import { mapGetters, mapActions } from "vuex";
+import { PracticeMode, Difficulty } from "../engine/models/math_question";
+
+import BasicDiffIcon from "../assets/difficulty_icons/basic.svg"
```
Comment (ArsalaBangash) [inline src/components/PracticeCard.vue:66]: I think BasicIcon gets the point across

File: src/components/PracticeCard.vue [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+import NormalDiffIcon from "../assets/difficulty_icons/normal.svg"
+import AdvancedDiffIcon from "../assets/difficulty_icons/advanced.svg"
+
+export default {
+  props: {
+    icon: {
```
Comment (ArsalaBangash) [inline src/components/PracticeCard.vue:72]: Is this prop being used?

**Review by ArsalaBangash — CHANGES_REQUESTED** [review]
## Solid iteration

Great work! Here are some points I have:

- [x] Realign the spacing in the customize practice card

### How it was before

image


### How it is now

image


- [x] We use the kebab-case notation for directories, so let's rename assets//difficulty_icons to assets/difficulty-icons/
- [x] I think CustomizePracticeCard or CustomizePractice would be a better name for the component since PracticeCard implies it's a card component where the practice occurs

File: src/components/PracticeCard.vue (hunk 1 above)
Comment (diracs-delta) [reply src/components/PracticeCard.vue:66]: done

File: src/components/PracticeCard.vue (hunk 2 above)
Comment (diracs-delta) [reply src/components/PracticeCard.vue:72]: nope. removed

**Review by ArsalaBangash — APPROVED** [review]
Great work!

**Pull request merged**